File: redome/src/main/java/br/org/cancer/modred/controller/dto/MatchDTO.java
package br.org.cancer.modred.controller.dto;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

import br.org.cancer.modred.model.StatusDoador;
import br.org.cancer.modred.model.TipoPermissividade;

/**
 * Classe de DTO para informações do match.
 * 
 * @author Fillipe Queiroz
 */
public class MatchDTO {

	private Long id;
	private Long idDoador;
	private String idRegistro;
	private String registroOrigem;
	private Long dmr;
	private String nome;
	private String sexo;
	private LocalDate dataNascimento;
	private LocalDateTime dataAtualizacao;
	private BigDecimal peso;
	private String abo;
	private String mismatch;
	private int outrosProcessos;
	private Boolean possuiRessalva;
	private Boolean possuiComentario;
	private String classificacao;
	private Boolean temPrescricao;
	private Long idProcesso;
	private Long idPedidoExame;
	private Long idTarefa;
	private Long idTipoTarefa;
	private Long idSolicitacao;
	private BigDecimal quantidadeTCNPorKilo;
	private BigDecimal quantidadeCD34PorKilo;
	private Long tipoDoador;
	private String idBscup;
	private String locusPedidoExameParaPaciente;
	private String locusPedidoExame;
	private Long rmrPedidoExame;
	private Boolean possuiGenotipoDivergente;
	private String fase;
	private String tipoMatch;
	private Boolean disponibilizado;
	private Long tipoSolicitacao;
	private StatusDoador statusDoador;
	private Long respostaQtdGestacaoDoador;
	private Integer somaPesoQualificacao;
	private TipoPermissividade tipoPermissividade;
	private Integer ordenacaoWmdaMatch;
	
	private List<BuscaChecklistDTO> buscaChecklist;
	
	/**
	 * @return the id
	 */
	public Long getId() {
		return id;
	}

	/**
	 * @param id the id to set
	 */
	public void setId(Long id) {
		this.id = id;
	}

	/**
	 * @return the idDoador
	 */
	public Long getIdDoador() {
		return idDoador;
	}

	/**
	 * @param idDoador the idDoador to set
	 */
	public void setIdDoador(Long idDoador) {
		this.idDoador = idDoador;
	}

	/**
	 * @return the nome
	 */
	public String getNome() {
		return nome;
	}

	/**
	 * @param nome the nome to set
	 */
	public void setNome(String nome) {
		this.nome = nome;
	}

	/**
	 * @return the sexo
	 */
	public String getSexo() {
		return sexo;
	}

	/**
	 * @param sexo the sexo to set
	 */
	public void setSexo(String sexo) {
		this.sexo = sexo;
	}

	/**
	 * @return the dataNascimento
	 */
	public LocalDate getDataNascimento() {
		return dataNascimento;
	}

	/**
	 * @param dataNascimento the dataNascimento to set
	 */
	public void setDataNascimento(LocalDate dataNascimento) {
		this.dataNascimento = dataNascimento;
	}

	/**
	 * @return the peso
	 */
	public BigDecimal getPeso() {
		return peso;
	}

	/**
	 * @param peso the peso to set
	 */
	public void setPeso(BigDecimal peso) {
		this.peso = peso;
	}

	/**
	 * @return the abo
	 */
	public String getAbo() {
		return abo;
	}

	/**
	 * @param abo the abo to set
	 */
	public void setAbo(String abo) {
		this.abo = abo;
	}

	/**
	 * @return the mismatch
	 */
	public String getMismatch() {
		return mismatch;
	}

	/**
	 * @param mismatch the mismatch to set
	 */
	public void setMismatch(String mismatch) {
		this.mismatch = mismatch;
	}

	/**
	 * Quantidade de processos envolvidos pelo doador.
	 * @return
	 */
	public int getOutrosProcessos() {
		return outrosProcessos;
	}

	/**
	 * @param outrosProcessos the outrosProcessos to set
	 */
	public void setOutrosProcessos(int outrosProcessos) {
		this.outrosProcessos = outrosProcessos;
	}

	/**
	 * @return the possuiRessalva
	 */
	public Boolean getPossuiRessalva() {
		return possuiRessalva;
	}

	/**
	 * @param possuiRessalva the possuiRessalva to set
	 */
	public void setPossuiRessalva(Boolean possuiRessalva) {
		this.possuiRessalva = possuiRessalva;
	}

	/**
	 * @return the possuiComentario
	 */
	public Boolean getPossuiComentario() {
		return possuiComentario;
	}

	/**
	 * @param possuiComentario the possuiComentario to set
	 */
	public void setPossuiComentario(Boolean possuiComentario) {
		this.possuiComentario = possuiComentario;
	}

	/**
	 * @return the classificacao
	 */
	public String getClassificacao() {
		return classificacao;
	}

	/**
	 * @param classificacao the classificacao to set
	 */
	public void setClassificacao(String classificacao) {
		this.classificacao = classificacao;
	}

	/**
	 * @return the dataAtualizacao
	 */
	public LocalDateTime getDataAtualizacao() {
		return dataAtualizacao;
	}

	/**
	 * @param dataAtualizacao the dataAtualizacao to set
	 */
	public void setDataAtualizacao(LocalDateTime dataAtualizacao) {
		this.dataAtualizacao = dataAtualizacao;
	}

	/**
	 * @return the temPrescricao
	 */
	public Boolean getTemPrescricao() {
		return temPrescricao;
	}

	/**
	 * @param temPrescricao the temPrescricao to set
	 */
	public void setTemPrescricao(Boolean temPrescricao) {
		this.temPrescricao = temPrescricao;
	}

	/**
	 * @return the dmr
	 */
	public Long getDmr() {
		return dmr;
	}

	/**
	 * @param dmr the dmr to set
	 */
	public void setDmr(Long dmr) {
		this.dmr = dmr;
	}

	/**
	 * @return the idRegistro
	 */
	public String getIdRegistro() {
		return idRegistro;
	}

	/**
	 * @param idRegistro the idRegistro to set
	 */
	public void setIdRegistro(String idRegistro) {
		this.idRegistro = idRegistro;
	}

	/**
	 * @return the registroOrigem
	 */
	public String getRegistroOrigem() {
		return registroOrigem;
	}

	/**
	 * @param registroOrigem the registroOrigem to set
	 */
	public void setRegistroOrigem(String registroOrigem) {
		this.registroOrigem = registroOrigem;
	}

	/**
	 * @return the idProcesso
	 */
	public Long getIdProcesso() {
		return idProcesso;
	}

	/**
	 * @param idProcesso the idProcesso to set
	 */
	public void setIdProcesso(Long idProcesso) {
		this.idProcesso = idProcesso;
	}

	/**
	 * @return the idPedidoExame
	 */
	public Long getIdPedidoExame() {
		return idPedidoExame;
	}

	/**
	 * @param idPedidoExame the idPedidoExame to set
	 */
	public void setIdPedidoExame(Long idPedidoExame) {
		this.idPedidoExame = idPedidoExame;
	}

	/**
	 * @return the idTarefa
	 */
	public Long getIdTarefa() {
		return idTarefa;
	}

	/**
	 * @param idTarefa the idTarefa to set
	 */
	public void setIdTarefa(Long idTarefa) {
		this.idTarefa = idTarefa;
	}

	/**
	 * @return the idTipoTarefa
	 */
	public Long getIdTipoTarefa() {
		return idTipoTarefa;
	}

	/**
	 * @param idTipoTarefa the idTipoTarefa to set
	 */
	public void setIdTipoTarefa(Long idTipoTarefa) {
		this.idTipoTarefa = idTipoTarefa;
	}

	/**
	 * @return the idSolicitacao
	 */
	public Long getIdSolicitacao() {
		return idSolicitacao;
	}

	/**
	 * @param idSolicitacao the idSolicitacao to set
	 */
	public void setIdSolicitacao(Long idSolicitacao) {
		this.idSolicitacao = idSolicitacao;
	}

	
	/**
	 * @return the quantidadeTCNPorKilo
	 */
	public BigDecimal getQuantidadeTCNPorKilo() {
		return quantidadeTCNPorKilo;
	}

	
	/**
	 * @param quantidadeTCNPorKilo the quantidadeTCNPorKilo to set
	 */
	public void setQuantidadeTCNPorKilo(BigDecimal quantidadeTCNPorKilo) {
		this.quantidadeTCNPorKilo = quantidadeTCNPorKilo;
	}
	
	/**
	 * @return the quantidadeCD34PorKilo
	 */
	public BigDecimal getQuantidadeCD34PorKilo() {
		return quantidadeCD34PorKilo;
	}
	
	/**
	 * @param quantidadeCD34PorKilo the quantidadeCD34PorKilo to set
	 */
	public void setQuantidadeCD34PorKilo(BigDecimal quantidadeCD34PorKilo) {
		this.quantidadeCD34PorKilo = quantidadeCD34PorKilo;
	}

	/**
	 * @return the tipoDoador
	 */
	public Long getTipoDoador() {
		return tipoDoador;
	}
	
	/**
	 * @param tipoDoador the tipoDoador to set
	 */
	public void setTipoDoador(Long tipoDoador) {
		this.tipoDoador = tipoDoador;
	}

	
	/**
	 * @return the idBscup
	 */
	public String getIdBscup() {
		return idBscup;
	}

	
	/**
	 * @param idBscup the idBscup to set
	 */
	public void setIdBscup(String idBscup) {
		this.idBscup = idBscup;
	}

	/**
	 * @return the locusPedidoExameParaPaciente
	 */
	public String getLocusPedidoExameParaPaciente() {
		return locusPedidoExameParaPaciente;
	}

	/**
	 * @param locusPedidoExameParaPaciente the locusPedidoExameParaPaciente to set
	 */
	public void setLocusPedidoExameParaPaciente(String locusPedidoExameParaPaciente) {
		this.locusPedidoExameParaPaciente = locusPedidoExameParaPaciente;
	}

	/**
	 * @return the locusPedidoExame
	 */
	public String getLocusPedidoExame() {
		return locusPedidoExame;
	}

	/**
	 * @param locusPedidoExame the locusPedidoExame to set
	 */
	public void setLocusPedidoExame(String locusPedidoExame) {
		this.locusPedidoExame = locusPedidoExame;
	}

	/**
	 * @return the rmrPedidoExame
	 */
	public Long getRmrPedidoExame() {
		return rmrPedidoExame;
	}

	/**
	 * @param rmrPedidoExame the rmrPedidoExame to set
	 */
	public void setRmrPedidoExame(Long rmrPedidoExame) {
		this.rmrPedidoExame = rmrPedidoExame;
	}

	/**
	 * Lista de itens de checklist a serem exibidos para o doador.
	 * 
	 * @return lista de DTOs contendo a descrição e prazo do ocorrido.
	 */
	public List<BuscaChecklistDTO> getBuscaChecklist() {
		return buscaChecklist;
	}

	public void setBuscaChecklist(List<BuscaChecklistDTO> buscaChecklist) {
		this.buscaChecklist = buscaChecklist;
	}

	/**
	 * @return the possuiGenotipoDivergente
	 */
	public Boolean getPossuiGenotipoDivergente() {
		return possuiGenotipoDivergente;
	}

	/**
	 * @param possuiGenotipoDivergente the possuiGenotipoDivergente to set
	 */
	public void setPossuiGenotipoDivergente(Boolean possuiGenotipoDivergente) {
		this.possuiGenotipoDivergente = possuiGenotipoDivergente;
	}

	/**
	 * @return the fase
	 */
	public String getFase() {
		return fase;
	}

	/**
	 * @param fase the fase to set
	 */
	public void setFase(String fase) {
		this.fase = fase;
	}

	/**
	 * @return the tipoMatch
	 */
	public String getTipoMatch() {
		return tipoMatch;
	}

	/**
	 * @param tipoMatch the tipoMatch to set
	 */
	public void setTipoMatch(String tipoMatch) {
		this.tipoMatch = tipoMatch;
	}

	/**
	 * @return the disponibilizado
	 */
	public Boolean getDisponibilizado() {
		return disponibilizado;
	}

	/**
	 * @param disponibilizado the disponibilizado to set
	 */
	public void setDisponibilizado(Boolean disponibilizado) {
		this.disponibilizado = disponibilizado;
	}

	/**
	 * @return the tipoSolicitacao
	 */
	public Long getTipoSolicitacao() {
		return tipoSolicitacao;
	}

	/**
	 * @param tipoSolicitacao the tipoSolicitacao to set
	 */
	public void setTipoSolicitacao(Long tipoSolicitacao) {
		this.tipoSolicitacao = tipoSolicitacao;
	}

	/**
	 * @return the somaPesoQualificacao
	 */
	public Integer getSomaPesoQualificacao() {
		return somaPesoQualificacao;
	}

	/**
	 * @param somaPesoQualificacao the somaPesoQualificacao to set
	 */
	public void setSomaPesoQualificacao(Integer somaPesoQualificacao) {
		this.somaPesoQualificacao = somaPesoQualificacao;
	}	

	/**
	 * @return the statusDoador
	 */
	public StatusDoador getStatusDoador() {
		return statusDoador;
	}

	/**
	 * @param statusDoador the statusDoador to set
	 */
	public void setStatusDoador(StatusDoador statusDoador) {
		this.statusDoador = statusDoador;
	}

	/**
	 * @return the respostaQtdGestacaoDoador
	 */
	public Long getRespostaQtdGestacaoDoador() {
		return respostaQtdGestacaoDoador;
	}

	/**
	 * @param respostaQtdGestacaoDoador the respostaQtdGestacaoDoador to set
	 */
	public void setRespostaQtdGestacaoDoador(Long respostaQtdGestacaoDoador) {
		this.respostaQtdGestacaoDoador = respostaQtdGestacaoDoador;
	}

	/**
	 * @return the tipoPermissividade
	 */
	public TipoPermissividade getTipoPermissividade() {
		return tipoPermissividade;
	}

	/**
	 * @param tipoPermissividade the tipoPermissividade to set
	 */
	public void setTipoPermissividade(TipoPermissividade tipoPermissividade) {
		this.tipoPermissividade = tipoPermissividade;
	}

	/**
	 * @return the ordenacaoWmdaMatch
	 */
	public Integer getOrdenacaoWmdaMatch() {
		return ordenacaoWmdaMatch;
	}

	/**
	 * @param ordenacaoWmdaMatch the ordenacaoWmdaMatch to set
	 */
	public void setOrdenacaoWmdaMatch(Integer ordenacaoWmdaMatch) {
		this.ordenacaoWmdaMatch = ordenacaoWmdaMatch;
	}
	
}

File: redome-front/src/app/shared/component/listamatch/match.dto.ts
import { StatusDoador } from './../../dominio/status.doador';
import { ConvertUtil } from 'app/shared/util/convert.util';
import { BaseEntidade } from '../../base.entidade';
import { TiposDoador } from '../../enums/tipos.doador';
import { BuscaChecklistDTO } from './classes/busca.checklist.dto';
import { IFiltrosOrdenacaoBuscaDTO } from 'app/shared/dto/interface.filtros.ordenacao.busca.dto';
import { GenotipoDTO } from 'app/paciente/genotipo.dto';


/**
 * @description DTO que representa um item no carrousel de matchs.
 *
 * @author Pizão
 * @export
 * @class MatchDTO
 * @extends {BaseEntidade}
 */
export class MatchDTO extends BaseEntidade implements IFiltrosOrdenacaoBuscaDTO {


	private _id: number;
	private _idDoador: number;
    private _dmr: number;
	private _nome: string;
	private _registroOrigem: string;
    private _sexo: string;
    private _dataNascimento: Date;
    private _peso: number;
    private _abo: string;
    private _mismatch: string;
    private _classificacao: string;
	private _outrosProcessos: number;
	private _possuiRessalva: boolean;
	private _possuiComentario: boolean = false;
	private _ehFavorito:boolean = false;
	private _dataAtualizacao: Date;
	private _idRegistro: string;

	private _temPrescricao: boolean;

	private _idProcesso: number;
	private _idPedidoExame: number;
	private _idTarefa: number;
	private _idTipoTarefa:number;
	private _idSolicitacao:number;

	private _quantidadeTCNPorKilo: number;
	private _quantidadeCD34PorKilo: number;
	private _tipoDoador: TiposDoador;
	private _idBscup: string;

	private _buscaChecklist: BuscaChecklistDTO[];

	private _locusPedidoExameParaPaciente:string;
	private _locusPedidoExame: string;
	private _rmrPedidoExame:number;

	private _possuiGenotipoDivergente: boolean;

	private _fase: string;
	private _disponibilizado: boolean;
	private _tipoSolicitacao: number;

	private _genotipoDoador: GenotipoDTO[];
	private _respostaQtdGestacaoDoador: number;
	private _somaQualificacao: number;
  private _statusDoador: StatusDoador;

  private _ordenacaoWmdaMatch: number;

	public get id(): number {
		return this._id;
	}

	public set id(value: number) {
		this._id = value;
	}

	public get nome(): string {
		return this._nome;
	}

	public set nome(value: string) {
		this._nome = value;
	}

    /**
     * Getter idDoador
     * @return {number}
     */
	public get idDoador(): number {
		return this._idDoador;
	}

    /**
     * Setter idDoador
     * @param {number} value
     */
	public set idDoador(value: number) {
		this._idDoador = value;
	}


	public get dmr(): number {
		return this._dmr;
	}

	public set dmr(value: number) {
		this._dmr = value;
	}

	public set sexo(value: string) {
		this._sexo = value;
	}

	public get sexo(): string {
		return this._sexo;
	}

	public get dataNascimento(): Date {
		return this._dataNascimento;
	}

	public set dataNascimento(value: Date) {
		this._dataNascimento = value;
	}

	public get peso(): number {
		return this._peso;
	}

	public set peso(value: number) {
		this._peso = value;
	}

	public get abo(): string {
		return this._abo;
	}

	public set abo(value: string) {
		this._abo = value;
	}

	public get mismatch(): string {
		return this._mismatch;
	}

	public set mismatch(value: string) {
		this._mismatch = value;
	}

	public get classificacao(): string {
		return this._classificacao;
	}

	public set classificacao(value: string) {
		this._classificacao = value;
    }

	public get outrosProcessos(): number {
		return this._outrosProcessos;
	}

	public set outrosProcessos(value: number) {
		this._outrosProcessos = value;
	}


	public get possuiRessalva(): boolean {
		return this._possuiRessalva;
	}

	public set possuiRessalva(value: boolean) {
		this._possuiRessalva = value;
	}

	public get possuiComentario(): boolean {
		return this._possuiComentario;
	}

	public set possuiComentario(value: boolean) {
		this._possuiComentario = value;
	}
	public get ehFavorito(): boolean {
		return this._ehFavorito;
	}

	public set ehFavorito(value: boolean) {
		this._ehFavorito = value;
	}
	public get dataAtualizacao(): Date {
		return this._dataAtualizacao;
	}

	public set dataAtualizacao(value: Date) {
		this._dataAtualizacao = value;
	}


	public get temPrescricao(): boolean {
		return this._temPrescricao;
	}

	public set temPrescricao(value: boolean) {
		this._temPrescricao = value;
	}

    /**
     * Getter idRegistro
     * @return {string}
     */
	public get idRegistro(): string {
		return this._idRegistro;
	}

    /**
     * Setter idRegistro
     * @param {string} value
     */
	public set idRegistro(value: string) {
		this._idRegistro = value;
	}

    /**
     * Getter registroOrigem
     * @return {string}
     */
	public get registroOrigem(): string {
		return this._registroOrigem;
	}

    /**
     * Setter registroOrigem
     * @param {string} value
     */
	public set registroOrigem(value: string) {
		this._registroOrigem = value;
	}


    /**
     * Getter idProcesso
     * @return {number}
     */
	public get idProcesso(): number {
		return this._idProcesso;
	}

    /**
     * Setter idProcesso
     * @param {number} value
     */
	public set idProcesso(value: number) {
		this._idProcesso = value;
	}

    /**
     * Getter idPedidoExame
     * @return {number}
     */
	public get idPedidoExame(): number {
		return this._idPedidoExame;
	}

    /**
     * Setter idPedidoExame
     * @param {number} value
     */
	public set idPedidoExame(value: number) {
		this._idPedidoExame = value;
	}

    /**
     * Getter idTarefa
     * @return {number}
     */
	public get idTarefa(): number {
		return this._idTarefa;
	}

    /**
     * Setter idTarefa
     * @param {number} value
     */
	public set idTarefa(value: number) {
		this._idTarefa = value;
	}

    /**
     * Getter idTipoTarefa
     * @return {number}
     */
	public get idTipoTarefa(): number {
		return this._idTipoTarefa;
	}

    /**
     * Setter idTipoTarefa
     * @param {number} value
     */
	public set idTipoTarefa(value: number) {
		this._idTipoTarefa = value;
	}

    /**
     * Getter idSolicitacao
     * @return {number}
     */
	public get idSolicitacao(): number {
		return this._idSolicitacao;
	}

    /**
     * Setter idSolicitacao
     * @param {number} value
     */
	public set idSolicitacao(value: number) {
		this._idSolicitacao = value;
	}

    /**
     * Getter quantidadeTCNPorKilo
     * @return {number}
     */
	public get quantidadeTCNPorKilo(): number {
		return this._quantidadeTCNPorKilo;
	}

    /**
     * Setter quantidadeTCNPorKilo
     * @param {number} value
     */
	public set quantidadeTCNPorKilo(value: number) {
		this._quantidadeTCNPorKilo = value;
	}

    /**
     * Getter quantidadeCD34PorKilo
     * @return {number}
     */
	public get quantidadeCD34PorKilo(): number {
		return this._quantidadeCD34PorKilo;
	}

    /**
     * Setter quantidadeCD34PorKilo
     * @param {number} value
     */
	public set quantidadeCD34PorKilo(value: number) {
		this._quantidadeCD34PorKilo = value;
	}

    /**
     * Getter tipoDoador
     * @return {TiposDoador}
     */
	public get tipoDoador(): TiposDoador {
		return this._tipoDoador;
	}

    /**
     * Setter tipoDoador
     * @param {TiposDoador} value
     */
	public set tipoDoador(value: TiposDoador) {
		this._tipoDoador = value;
	}


	/**
	 * Verifica se o match é correspondente a
	 * medula.
	 * @return TRUE se for medula.
	 */
	isMedula(): boolean{
		return TiposDoador.NACIONAL == this.tipoDoador
					|| TiposDoador.INTERNACIONAL == this.tipoDoador;
	}

	/**
	 * Verifica se o match é correspondente a
	 * cordao.
	 * @return TRUE se for cordao.
	 */
	isCordao(): boolean{
		return TiposDoador.CORDAO_NACIONAL == this.tipoDoador
					|| TiposDoador.CORDAO_INTERNACIONAL == this.tipoDoador;
	}

	/**
	 * @description Indica se o match possui itens de checklist
	 * que o analista de busca precisa tomar ciência e dar como visto.
	 * @type {BuscaChecklistDTO[]}
	 */
	public get buscaChecklist(): BuscaChecklistDTO[] {
		return this._buscaChecklist;
	}

	public set buscaChecklist(value: BuscaChecklistDTO[]) {
		this._buscaChecklist = value;
	}


    /**
     * Getter locusPedidoExameParaPaciente
     * @return {string}
     */
	public get locusPedidoExameParaPaciente(): string {
		return this._locusPedidoExameParaPaciente;
	}

    /**
     * Setter locusPedidoExameParaPaciente
     * @param {string} value
     */
	public set locusPedidoExameParaPaciente(value: string) {
		this._locusPedidoExameParaPaciente = value;
	}

    /**
     * Getter locusPedidoExame
     * @return {string}
     */
	public get locusPedidoExame(): string {
		return this._locusPedidoExame;
	}

    /**
     * Setter locusPedidoExame
     * @param {string} value
     */
	public set locusPedidoExame(value: string) {
		this._locusPedidoExame = value;
	}

    /**
     * Getter rmrPedidoExame
     * @return {number}
     */
	public get rmrPedidoExame(): number {
		return this._rmrPedidoExame;
	}

    /**
     * Setter rmrPedidoExame
     * @param {number} value
     */
	public set rmrPedidoExame(value: number) {
		this._rmrPedidoExame = value;
	}

    /**
     * Getter possuiGenotipoDivergente
     * @return {boolean}
     */
	public get possuiGenotipoDivergente(): boolean {
		return this._possuiGenotipoDivergente;
	}

    /**
     * Setter possuiGenotipoDivergente
     * @param {boolean} value
     */
	public set possuiGenotipoDivergente(value: boolean) {
		this._possuiGenotipoDivergente = value;
	}

    /**
     * Getter fase
     * @return {string}
     */
	public get fase(): string {
		return this._fase;
	}

    /**
     * Setter fase
     * @param {string} value
     */
	public set fase(value: string) {
		this._fase = value;
	}

    /**
     * Getter disponibilizado
     * @return {boolean}
     */
	public get disponibilizado(): boolean {
		return this._disponibilizado;
	}

    /**
     * Setter disponibilizado
     * @param {boolean} value
     */
	public set disponibilizado(value: boolean) {
		this._disponibilizado = value;
	}

    /**
     * Getter tipoSolicitacao
     * @return {number}
     */
	public get tipoSolicitacao(): number {
		return this._tipoSolicitacao;
	}

    /**
     * Setter tipoSolicitacao
     * @param {number} value
     */
	public set tipoSolicitacao(value: number) {
		this._tipoSolicitacao = value;
	}


    /**
     * Getter idBscup
     * @return {string}
     */
	public get idBscup(): string {
		return this._idBscup;
	}

    /**
     * Setter idBscup
     * @param {string} value
     */
	public set idBscup(value: string) {
		this._idBscup = value;
	}

	/**
     * Getter genotipoDoador
     * @return {GenotipoDTO[]}
     */
	public get genotipoDoador(): GenotipoDTO[] {
		return this._genotipoDoador;
	}

    /**
     * Setter genotipoDoador
     * @param {GenotipoDTO[]} value
     */
	public set genotipoDoador(value: GenotipoDTO[]) {
		this._genotipoDoador = value;
	}

    /**
     * Getter somaQualificacao
     * @return {number}
     */
	public get somaQualificacao(): number {
		return this._somaQualificacao;
	}

    /**
     * Setter somaQualificacao
     * @param {number} value
     */
	public set somaQualificacao(value: number) {
		this._somaQualificacao = value;
	}

	/**
     * Getter statusDoador
     * @return {StatusDoador}
     */
	public get statusDoador(): StatusDoador {
		return this._statusDoador;
	}

	/**
     * Setter statusDoador
     * @param {StatusDoador} value
     */
	public set statusDoador(value: StatusDoador) {
		this._statusDoador = value;
	}

    /**
     * Getter respostaQtdGestacaoDoador
     * @return {number}
     */
	public get respostaQtdGestacaoDoador(): number {
		return this._respostaQtdGestacaoDoador;
	}

    /**
     * Setter respostaQtdGestacaoDoador
     * @param {number} value
     */
	public set respostaQtdGestacaoDoador(value: number) {
		this._respostaQtdGestacaoDoador = value;
	}


    /**
     * Getter ordenacaoWmdaMatch
     * @return {number}
     */
	public get ordenacaoWmdaMatch(): number {
		return this._ordenacaoWmdaMatch;
	}

    /**
     * Setter ordenacaoWmdaMatch
     * @param {number} value
     */
	public set ordenacaoWmdaMatch(value: number) {
		this._ordenacaoWmdaMatch = value;
	}


	public jsonToEntity(res: any): MatchDTO {

		if (res.tipoDoador != null) {
			this.tipoDoador = TiposDoador.valueOf(res.tipoDoador);
			//let idTipoDoador: string = new String(res.tipoDoador).valueOf();
			//this.tipoDoador = TiposDoador[idTipoDoador as keyof typeof TiposDoador];
		}

		this.statusDoador = new StatusDoador().jsonToEntity(res);

		if (res.buscaChecklist) {
			this.buscaChecklist = [];
			res.buscaChecklist.forEach(check => {
				this.buscaChecklist.push(new BuscaChecklistDTO().jsonToEntity(check));
			})
		}

		this.id = ConvertUtil.parseJsonParaAtributos(res.id , new Number());
		this.idDoador = ConvertUtil.parseJsonParaAtributos(res.idDoador , new Number());
		this.dmr = ConvertUtil.parseJsonParaAtributos(res.dmr , new Number());
		this.nome = ConvertUtil.parseJsonParaAtributos(res.nome , new String());
		this.registroOrigem = ConvertUtil.parseJsonParaAtributos(res.registroOrigem , new String());
		this.sexo = ConvertUtil.parseJsonParaAtributos(res.sexo , new String());
		this.dataNascimento = ConvertUtil.parseJsonParaAtributos(res.dataNascimento , new Date());
		this.peso = ConvertUtil.parseJsonParaAtributos(res.peso , new Number());
		this.abo = ConvertUtil.parseJsonParaAtributos(res.abo , new String());
		this.mismatch = ConvertUtil.parseJsonParaAtributos(res.mismatch , new String());
		this.classificacao = ConvertUtil.parseJsonParaAtributos(res.classificacao , new String());
		this.outrosProcessos = ConvertUtil.parseJsonParaAtributos(res.outrosProcessos , new Number());
		this.possuiRessalva = ConvertUtil.parseJsonParaAtributos(res.possuiRessalva , new Boolean());
		this.possuiComentario = ConvertUtil.parseJsonParaAtributos(res.possuiComentario , new Boolean());
		this.ehFavorito = ConvertUtil.parseJsonParaAtributos(res.ehFavorito , new Boolean());
		this.dataAtualizacao = ConvertUtil.parseJsonParaAtributos(res.dataAtualizacao , new Date());
		this.idRegistro = ConvertUtil.parseJsonParaAtributos(res.idRegistro , new String());

		this.temPrescricao = ConvertUtil.parseJsonParaAtributos(res.temPrescricao , new Boolean());

		this.idProcesso = ConvertUtil.parseJsonParaAtributos(res.idProcesso , new Number());
		this.idPedidoExame = ConvertUtil.parseJsonParaAtributos(res.idPedidoExame , new Number());
		this.idTarefa = ConvertUtil.parseJsonParaAtributos(res.idTarefa , new Number());
		this.idTipoTarefa = ConvertUtil.parseJsonParaAtributos(res.idTipoTarefa , new Number());
		this.idSolicitacao = ConvertUtil.parseJsonParaAtributos(res.idSolicitacao , new Number());

		this.quantidadeTCNPorKilo = ConvertUtil.parseJsonParaAtributos(res.quantidadeTCNPorKilo , new Number());
		this.quantidadeCD34PorKilo = ConvertUtil.parseJsonParaAtributos(res.quantidadeCD34PorKilo , new Number());
		this.idBscup = ConvertUtil.parseJsonParaAtributos(res.idBscup , new String());
		this.locusPedidoExame = ConvertUtil.parseJsonParaAtributos(res.locusPedidoExame , new String());
		this.locusPedidoExameParaPaciente = ConvertUtil.parseJsonParaAtributos(res.locusPedidoExameParaPaciente , new String());
		this.rmrPedidoExame = ConvertUtil.parseJsonParaAtributos(res.rmrPedidoExame , new Number());
		this.possuiGenotipoDivergente = ConvertUtil.parseJsonParaAtributos(res.possuiGenotipoDivergente, new Boolean());
		this.fase = ConvertUtil.parseJsonParaAtributos(res.fase, new String());
		this.disponibilizado = ConvertUtil.parseJsonParaAtributos(res.disponibilizado, new Boolean());
		this.tipoSolicitacao = ConvertUtil.parseJsonParaAtributos(res.tipoSolicitacao, new Number());
		this.genotipoDoador = res.genotipoDoadores? res.genotipoDoadores: null;
		this.somaQualificacao = ConvertUtil.parseJsonParaAtributos(res.somaPesoQualificacao, new Number());
		this.statusDoador = ConvertUtil.parseJsonParaAtributos(res.statusDoador, new String());
		this.respostaQtdGestacaoDoador = ConvertUtil.parseJsonParaAtributos(res.respostaQtdGestacaoDoador, new Number());
    this.ordenacaoWmdaMatch = ConvertUtil.parseJsonParaAtributos(res.ordenacaoWmdaMatch, new Number());

        return this;
    }

}
